modules/packages/LinearAlgebra.chpl

```diff
@@ -503,8 +503,44 @@ proc transpose(A: [?Dom] ?eltType) where isDenseMatrix(A) {
   }
 }
 
+pragma "no doc"
+proc tranpose(A: [?Dom] ?eltType, keepDistribution = true) 
+  where isDistributed(A) 
+{
+  // Add case for rank 1
+  const transposedDom = {Dom.dim(1), Dom.dim(0)};
+  const targetLocales = A.targetLocales();
+
+  if keepDistribution {
+    var distDom = transposedDom dmapped Block(boundingBox=transposedDom, 
+                                              targetLocales=targetLocales);
+    var transposedMat : [distDom] eltType;
+
+    coforall loc in targetLocales {
+      var localDomain = A.localSubdomain(); 
```

Comment:
Both out of principle and consistency, I'd make this a `const`.

FYI: we are working on finding ways of exploiting constness of domains to do some performance optimizations. I don't think it matters for this case, but in general there might be some benefits of using const domains in the future if that makes sense for the specific application.